### PR TITLE
chatty: init at 0.3.2

### PR DESCRIPTION
**Pull request by dotlambda**
###### Motivation for this change
continuation of https://github.com/NixOS/nixpkgs/pull/94820
I decided against the wrapper approach as that makes it hard to override other things and chatty doesn't take long to build. If a wrapper is preferred though, I'll change it of course.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [x] NixOS
   - [ ] macOS
   - [ ] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [ ] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

I still have to do extensive testing of all the plugins.

It would be nice if libpurple were separate from pidgin.

### CODE REVIEW

**Comment by dotlambda** [comment]
> For phosh/mobile usage, it might be good to include chatty with purple-mm-sms as a default package in the phosh module. Perhaps that can be saved for a future PR.

I was thinking of adding all supported plugins to the `plugins` list by default. If we add chatty to phosh's default packages, this would be especially nice because changing the plugins would require using an overlay rather than adding `chatty.override { plugins = ...; }` to `systemPackages`.

**Comment by Pacman99** [comment]
Thats a good idea. Make sure not to include `purple-matrix` the native matrix support in chatty is much better.

Also another consideration, should an override just append to the default plugin list or should it overwrite the plugin list entirely?

**Comment by dotlambda** [comment]
> Make sure not to include `purple-matrix` the native matrix support in chatty is much better.

How do I use the native one?

> Also another consideration, should an override just append to the default plugin list or should it overwrite the plugin list entirely?

I think overwriting it is the better choice, as people who override the package know which plugins exactly they want.

**Comment by dotlambda** [comment]
> > Make sure not to include `purple-matrix` the native matrix support in chatty is much better.
> 
> How do I use the native one?

I had to do `dconf write /sm/puri/Chatty/experimental-features true`.

**Comment by dotlambda** [comment]
Skimming through the code makes it seem like native matrix support is always used if experimental features are enabled, regarless of whether purple-matrix is installed. But I still have to confirm that.

**Comment by Pacman99** [comment]
> Skimming through the code makes it seem like native matrix support is always used if experimental features are enabled, regarless of whether purple-matrix is installed. But I still have to confirm that.

If thats true, then purple-matrix can just be included by default

>I think overwriting it is the better choice, as people who override the package know which plugins exactly they want.

That makes sense to me

**Comment by dotlambda** [comment]
I can't seem to get my Matrix account to work. I put in my data and it shows an account was created. However, my account keeps showing as disconnected and even using `--debug --verbose` doesn't produce any related error messages.

**Comment by tomfitzhenry** [comment]
FYI, https://github.com/NixOS/nixpkgs/pull/129230 is introducing `purple-mm-sms`.

**Comment by tomfitzhenry** [comment]
> I still have to do extensive testing of all the plugins.

It'd be great to get chatty in, even without all or any plugins (XMPP is included by default), to provide a Phosh-native chat client.

**Comment by dotlambda** [comment]
@tomfitzhenry If you can confirm that this is working, we can definitely add it. I just couldn't get it to work with my Matrix account and didn't bother with making an XMPP one.

**Comment by tomfitzhenry** [comment]
> @tomfitzhenry If you can confirm that this is working, we can definitely add it. I just couldn't get it to work with my Matrix account and didn't bother with making an XMPP one.

Done. I was able to login into an XMPP room and communicate, with chatty.

**Comment by tomfitzhenry** [comment]
I've been using this for 3 weeks now without issue. +1 to merging.

**Review by Pacman99 — APPROVED** [review]
LGTM

**Comment by dotlambda** [comment]
Who wants to step up to be chatty's maintainer? Since I can't test it, I shouldn't be.

**Comment by tomfitzhenry** [comment]
> Who wants to step up to be chatty's maintainer? Since I can't test it, I shouldn't be.

I am happy to be chatty's maintainer. I've been using this package for 7 weeks now, and am happy with the PR as-is. Please add me as the maintainer.

**Comment by dotlambda** [comment]
> > Who wants to step up to be chatty's maintainer? Since I can't test it, I shouldn't be.
> 
> I am happy to be chatty's maintainer. I've been using this package for 7 weeks now, and am happy with the PR as-is. Please add me as the maintainer.

Thanks! I left myself in so I'm still pinged when chatty is modified.
Would you mind updating the package to 0.3.4 after this is merged?

**Comment by dotlambda** [comment]
Result of `nixpkgs-review pr 122373` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>chatty</li>
  </ul>
</details>

**Review by tomfitzhenry — APPROVED** [review]

**Comment by tomfitzhenry** [comment]
> Would you mind updating the package to 0.3.4 after this is merged?

Yes, I will. I'm subscribed to https://source.puri.sm/Librem5/chatty/-/tags?format=atom too, for future releases.

**Comment by tomfitzhenry** [comment]
> > Would you mind updating the package to 0.3.4 after this is merged?
>
> Yes, I will.

https://github.com/NixOS/nixpkgs/pull/136527

**Pull request merged**
